server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -468,8 +469,13 @@ public Pool find(String poolId) {
             entitleDate, owner, null, fromPools);
 
         if (bestPools == null) {
-            throw new RuntimeException("No entitlements for products: " +
-                Arrays.toString(productIds));
+            List<String> fullList = new ArrayList<String>();
+            fullList.addAll(Arrays.asList(productIds));
+            for (ConsumerInstalledProduct cip : consumer.getInstalledProducts()) {
+                fullList.add(cip.getId());
+            }
+            log.warn("No entitlements available for products: " + fullList);
```

Comment:
The bind is across the installed products plus the ones passed to the method. This will make a complete list for the logging.